reporter/src/main/kotlin/utils/FreemarkerTemplateGenerator.kt

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright (C) 2020 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.reporter.utils
+
+import freemarker.cache.ClassTemplateLoader
+import freemarker.template.Configuration
+import freemarker.template.TemplateExceptionHandler
+
+import java.io.File
+
+import kotlin.reflect.full.memberProperties
+
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.licenses.LicenseConfiguration
+import org.ossreviewtoolkit.model.licenses.LicenseView
+import org.ossreviewtoolkit.model.licenses.ResolvedLicense
+import org.ossreviewtoolkit.model.licenses.ResolvedLicenseFileInfo
+import org.ossreviewtoolkit.model.licenses.ResolvedLicenseInfo
+import org.ossreviewtoolkit.model.licenses.filterExcluded
+import org.ossreviewtoolkit.reporter.ReporterInput
+import org.ossreviewtoolkit.utils.expandTilde
+import org.ossreviewtoolkit.utils.log
+
+class FreemarkerTemplateGenerator(
+    private val filePrefix: String,
+    private val defaultTemplateName: String
+) {
+    companion object {
+        private const val OPTION_TEMPLATE_ID = "template.id"
+        private const val OPTION_TEMPLATE_PATH = "template.path"
+    }
+
+    fun generateTemplate(input: ReporterInput, outputDir: File, options: Map<String, String>): List<File> {
+        val projects = input.ortResult.getProjects().map { project ->
+            PackageNoticeModel(project.id, input)
```

Comment:
Since this now it not specifically about NOTICE file generation anymore, I believe every mention of the substring "notice" in this file should be reviewed and replaced with something more generic.